docs/source/links/rest.md

```diff
@@ -0,0 +1,394 @@
+* _optional_ `endpoint?: string` key to use when looking up the endpoint in the (optional) `endpoints` table if provided to RestLink at creation time.
+* _optional_ `bodyKey?: string = "input"`: This is the name of the `variable` to use when looking to build a REST request-body for a `PUT` or `POST` request. It defaults to `input` if not supplied.
+* _optional_ `bodyBuilder?: /function/`: If provided, this is the name a `function` that you provided to `variables`, that is called when a request-body needs to be built. This lets you combine arguments or encode the body in some format other than JSON.
+
+<h3 id="rest.arguments.variables">Variables</h3>
+
+You can use query `variables` inside nested queries, or in the the path argument of your directive:
+
+```graphql
+query postTitle {
+  post(id: "1") @rest(type: "Post", path: "/post/:id") {
+    id
+    title
+  }
+}
+```
+
+<h4 id="rest.arguments.body">`bodyKey` / `bodyBuilder`</h4>
+
+When making a `POST` or `PUT` HTTP request, you often need to provide a request body. By [convention](https://graphql.org/graphql-js/mutations-and-input-types/), GraphQL recommends you name your input-types as `input`, so by default that's where we'll look to find a JSON object for your body.
+
+<h5 id="rest.arguments.body.key">`bodyKey`</h5>
+
+If you need/want to name it something different, you can pass `bodyKey`, and we'll look at that variable instead.
+
+In this example the publish API accepts a body in the variable `body` instead of input:
+
+```graphql
+mutation publishPost(
+  $someApiWithACustomBodyKey: PublishablePostInput!
+) {
+  publishedPost: publish(input: "Foo", body: $someApiWithACustomBodyKey)
+    @rest(
+      type: "Post"
+      path: "/posts/:input/new"
+      method: "POST"
+      bodyKey: "body"
+    ) {
+    id
+    title
+  }
+}
+```
+
+[Unit Test](https://github.com/apollographql/apollo-link-rest/blob/c9d81ae308e5f61b5ae992061de7abc6cb2f78e0/src/__tests__/restLink.ts#L1803-L1846)
+
+<h5 id="rest.arguments.body.builder">`bodyBuilder`</h5>
+
+If you need to structure your data differently, or you need to custom encode your body (say as form-encoded), you can instead provide `bodyBuilder`
+
+```graphql
+mutation encryptedPost(
+  $input: PublishablePostInput!
+  $encryptor: any
+) {
+  publishedPost: publish(input: $input)
+    @rest(
+      type: "Post"
+      path: "/posts/new"
+      method: "POST"
+      bodyBuilder: $encryptor
+    ) {
+    id
+    title
+  }
+}
+```
+
+[Unit Test](https://github.com/apollographql/apollo-link-rest/blob/c9d81ae308e5f61b5ae992061de7abc6cb2f78e0/src/__tests__/restLink.ts#L1847-L1904)
+
+<h2 id="export">@export directive</h2>
+
+The export directive re-exposes a field for use in a later (nested) query. These are the same semantics that will be supported on the server, but when used in a `RestLink` you can use the exported variables for further calls (i.e. waterfall requests from nested fields)
```

Comment:
I think we need a note about using a graphql server for more complicated rest control flows, which would allow caching across rest calls and client instances.

Reply:
Added!